scripts/prepare_sector_network.py

```diff
@@ -2032,20 +2037,21 @@ def add_services(n, costs):
         p_set=-co2,
     )
 
-    p_set_gas = (
-        profile_residential * energy_totals.loc[countries, "services gas"].sum() * 1e6
-    )
+    p_set_gas = 1e6 * profile_residential.mul(
+        energy_totals["services gas"], level=0

```

Comment:
What do you think: Does it make sense to write a method for setting p_set_gas, p_set_oil, p_set_biomass and p_set_elec as its often recurring?

something like
`def calculate_p_set(type):
        return 1e6 * profile_residential.mul(
        energy_totals[type], level=0
        ).droplevel(level=0, axis=1)
`

Reply:
Agree; done.
However, in the future we may consider restyling this approach.
To me, it seems that similar operations are repeated for multiple entities; we could create a for loop and automatize the operations for those terms for example.